src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    private BulldozerProducer bulldozerProducer;
+    private ExcavatorProducer excavatorProducer;
+    private TruckProducer truckProducer;
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            bulldozerProducer = new BulldozerProducer();
+            return new ArrayList<>(bulldozerProducer.get());
+        } else if (type.equals(Excavator.class)) {
+            excavatorProducer = new ExcavatorProducer();
+            return new ArrayList<>(excavatorProducer.get());
+        } else if (type.equals(Truck.class)) {
+            truckProducer = new TruckProducer();
+            return new ArrayList<>(truckProducer.get());
+        }
```

Comment:
https://github.com/mate-academy/hibernate-create-movie-session-hw/pull/344#pullrequestreview-1137964576
Do we need else after return in if clause? If not let’s not make our code longer unnecessary